src/test/java/com/google/devtools/build/lib/syntax/StarlarkEvaluationTest.java

```diff
@@ -1758,6 +1758,18 @@ public void testFunctionCallRecursion() throws Exception {
             "main()");
   }
 
+  @Test
+  public void testRecursionAllowedWithFlag() throws Exception {
+    new Scenario("--experimental_allow_recursion=true")
```

Comment:
The test will need to construct a StarlarkSemantics explicitly rather than through a flag, like so:
```
  @Test                                                                                                                                                                            
  public void testExec() throws Exception {                                                                                                                                        
    ParserInput input = ParserInput.fromLines("x = ...");
    Module module = Module.create();
    try (Mutability mu = Mutability.create("test")) {                                                                                                                              
      StarlarkThread thread = new StarlarkThread(mu, StarlarkSemantics...);
      EvalUtils.exec(input, FileOptions.DEFAULT, module, thread);                                                                                                                  
    }                                                                                                                                                                              
    assertThat(module.getGlobal("x")).isEqualTo(...);                                                                                           
  }                                                                                                                                                                                
```